### ISSUE TITLE
PMOS reverse protections 

### ISSUE THREAD

**Issue by Suke0811** [BerkeleyHCI/PolymorphicBlocks#323]
## Pmos

From 1.
![image](https://github.com/BerkeleyHCI/PolymorphicBlocks/assets/49264928/fbd38e90-540c-421c-8229-357b897949ab)

Implementation: 
https://github.com/BerkeleyHCI/PolymorphicBlocks/blob/5bb5b504e6a5da03bbd40ef9a66a422e02ea0a58/electronics_lib/PowerConditioning.py#L267-L305

## PMOS for bi-directional application (Low voltage)

From 2. 
![image](https://github.com/BerkeleyHCI/PolymorphicBlocks/assets/49264928/c3c48f34-081b-40e1-9b5b-c62365dc06b5)

Implementation: 

https://github.com/BerkeleyHCI/PolymorphicBlocks/blob/5bb5b504e6a5da03bbd40ef9a66a422e02ea0a58/electronics_lib/PowerConditioning.py#L310-L359

Currently, this code gets stuck during the netlist generations
![image](https://github.com/BerkeleyHCI/PolymorphicBlocks/assets/49264928/858cc309-92cb-445f-9738-904ead5e8d06)

## Ref
1.  https://components101.com/articles/design-guide-pmos-mosfet-for-reverse-voltage-polarity-protection
2.  https://www.edn.com/reverse-voltage-protection-for-battery-chargers/

**Comment by Suke0811** [comment]
- duplicated r1/r2 naming issues now detected with ducky update (5cbf405b4c51cfc1389c8e80515faf2c7b2c650b)

- Updated the charging protection to use schematics
chg and the power_out is manually connected, and the block looks like this?
![image](https://github.com/BerkeleyHCI/PolymorphicBlocks/assets/49264928/b330b22c-4ec5-4dbf-a93a-171d6371dc23)


```python
    chg_adapter = self.Block(PassiveAdapterVoltageSink())
    setattr(self, '(adapter)chg', chg_adapter)  # hack so the netlister recognizes this as an adapter
    self.connect(self.mp1.source, chg_adapter.src)
    self.connect(self.chg, chg_adapter.dst)
    ```

**Comment by ducky64** [comment]
Yeah, the diagram weirdness is from some weirdness in how the underlying layout libraries handles connections between exterior ports.

**Comment by Suke0811** [comment]
#328
